docs/guides/testing/testing.rst

```diff
@@ -117,144 +119,47 @@ Note -
-
-For the sake of speed, by default the python unit test database tables
-are created directly from apps' models. If you want to run the tests
-against a database created by applying the migrations instead, use the
-``--enable-migrations`` option::
-
-    paver test_system -s lms --enable-migrations
-
-To see the migration output, use::
-
-    paver test_system -s lms --enable-migrations --verbose --disable_capture
-
-To run a single django test class use this command::
-
-    paver test_system -t lms/djangoapps/courseware/tests/tests.py::ActivateLoginTest
 
 Running a Single Test
 ~~~~~~~~~~~~~~~~~~~~~
 
-When developing tests, it is often helpful to be able to really just run
-one single test without the overhead of PIP installs, UX builds, etc. In
-this case, it is helpful to look at the output of paver, and run just
-the specific command (optionally, stripping away coverage metrics). At
-the time of this writing, the command is the following::
-
-    pytest lms/djangoapps/courseware/tests/test_courses.py
-
-
-To run a single test format the command like this::
-
-    paver test_system -t lms/djangoapps/courseware/tests/tests.py::ActivateLoginTest::test_activate_login
-
-You can use ``--randomize`` to randomize the test case sequence.  In the
-short term, this is likely to reveal bugs in our test setup and teardown;
-please fix (or at least file tickets for) any such issues you encounter.
-
-You can also enable test concurrency with the ``--processes=N`` flag (where ``N``
-is the number of processes to run tests with, and ``-1`` means one process per
-available core). Note, however, that when running concurrently, breakpoints may
-not work correctly.
-
-For example::
-
-    # This will run all tests in the order that they appear in their files, serially
-    paver test_system -s lms --no-randomize --processes=0
-
-    # This will run using only 2 processes for tests
-    paver test_system -s lms --processes=2
+Various ways to run tests using pytest::
 
-To re-run all failing django tests from lms or cms, use the
-``--failed``,\ ``-f`` flag (see note at end of section)::
-
-    paver test_system -s lms --failed
-    paver test_system -s cms --failed
-
-There is also a ``--exitfirst``, ``-x`` option that will stop pytest
-after the first failure.
-
-common/lib tests are tested with the ``test_lib`` task, which also
-accepts the ``--failed`` and ``--exitfirst`` options::
-
-    paver test_lib -l common/lib/xmodule
-    paver test_lib -l common/lib/xmodule --failed
+    pytest path/test_m­odule.py                          # Run all tests in a module.
+    pytest path/test_m­odule.p­y:­:te­st_func               # Run a specific test within a module.
+    pytest path/test_m­odule.p­y:­:Te­stC­las­s               # Run all tests in a class
+    pytest path/test_m­odule.p­y:­:Te­stC­las­s::­tes­t_m­ethod  # Run a specific method of a class.
+    pytest path/testing/                                # Run all tests in a directory.
```

Comment:
Hmm, so i'm a little worried about missing in the context that python tests get run twice: once in the lms environment and once in the cms environment (I think, still). It is technically possible for tests that work in lms environments to fail in cms environments and vice versa. I'm not sure how best to include that information, and I'm not even sure the old documentation made it clear.

Reply:
I don't think old documentation made that clear, is there a way to run a test in lms environment and cms environment?

Reply:
There is a way to specify which environment to use: https://docs.pytest.org/en/stable/customize.html#initialization-determining-rootdir-and-configfile

I believe the lms settings are on by default and there's a specific configuration for cms: https://github.com/edx/edx-platform/blob/master/cms/pytest.ini

Reply:
@dianakhuang just made the changes, let me know what you think of my note